internal/services/consumption/consumption_budget_base.go

```diff
@@ -175,6 +175,12 @@ func (br consumptionBudgetBaseResource) arguments(fields map[string]*pluginsdk.S
 						}, false),
 					},
 
+					"locale": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringInSlice(budgets.PossibleValuesForCultureCode(), false),
+					},
```

Comment:
As this property is optional, we will need to specify a default value, or it will default to a blank string.

Reply:
How should this be implemented? If the default is used, the property is not returned by the api.
And if you want to use the default, the property must be removed from the post request.

Reply:
@stawik-mesa This feels like an API bug, but how we proceed probably depends on what the effective default value is. If this is omitted, does it behave as though it was set to `en-us`? If so, we should probably make this the default value. WDYT?

Reply:
The default value is inherited from the tenant. In my use case it's set to German and we want to make this configurable. e.g English, Spanish, .... so the only option at the moment is to not set the field at all to use the default value.

Reply:
@stawik-mesa Ah ok, thanks that makes sense. In that case, we can support an empty default value for this property, but it would be good to check whether we can unset the language once it has been set. If we can do that using a blank value then that's great, otherwise we might a conditional forcenew for this particular case (which can be implemented with a CustomizeDiff function). Let me know if you'd like some tips for working this out, or implementing it.

Reply:
@stawik-mesa apologies for the delayed response here, making this `Computed` until 4.0 (where we could then make this field Required) feels like the most sensible path forward given the API behaviour?

```suggestion
					"locale": {
						Type:         pluginsdk.TypeString,
						Optional:     true,
						Computed: true, // TODO: make this required in 4.0
						ValidateFunc: validation.StringInSlice(budgets.PossibleValuesForCultureCode(), false),
					},
```